activerecord/lib/active_record/associations.rb

```diff
@@ -1502,6 +1502,9 @@ module ClassMethods
         # [:ensuring_owner_was]
         #   Specifies an instance method to be called on the owner. The method must return true in order for the
         #   associated records to be deleted in a background job.
+        # [:query_constraints]
+        #   Serves as a composite foreign key. Defines the list of columns to be used to query the associated object.
+        #   The Array size must match associated model's primary key or `query_constraints` size.
```

Comment:
I think these docs (all the ones added here) should mention that this isn't required and Rails will try to guess what you want if not set.

Reply:
I added a generic note without mentioning the limitations of the automatic resolution logic. Let me know if you'd like to be more specific but a generic version allows Rails to always make the logic smarter without tweaking the docs